src/hotspot/share/gc/shared/referenceProcessor.hpp

```diff
@@ -596,42 +585,15 @@ class ReferenceProcessorAtomicMutator: StackObj {
   }
 };
 
-// This class is an interface used to implement task execution for the
-// reference processing.
-class AbstractRefProcTaskExecutor {
-public:
-
-  // Abstract tasks to execute.
-  class ProcessTask;
-
-  // Executes a task using worker threads.
-  virtual void execute(ProcessTask& task, uint ergo_workers) = 0;
-};
-
-// Abstract reference processing task to execute.
-class AbstractRefProcTaskExecutor::ProcessTask {
-protected:
-  ReferenceProcessor&           _ref_processor;
-  // Indicates whether the phase could generate work that should be balanced across
-  // threads after execution.
-  bool                          _marks_oops_alive;
-  ReferenceProcessorPhaseTimes* _phase_times;
-
-  ProcessTask(ReferenceProcessor& ref_processor,
-              bool marks_oops_alive,
-              ReferenceProcessorPhaseTimes* phase_times)
-    : _ref_processor(ref_processor),
-      _marks_oops_alive(marks_oops_alive),
-      _phase_times(phase_times)
-  { }
+enum class ThreadModel {Multi, Single};
```

Comment:
I am fine with adding the enum, but the name `ThreadModel`, at least as long as it is only used here, is too generic. Something like `RefProcThreadModel` would be just fine.

Also, please add spaces like this `{ Multi, Single }`

Reply:
I will rename and add spaces, I will remove the enum if another reviewer want it away.